Build failure on `txparse`

From a docker build, attempting `make -j8 txparse`:

```
#12 7.317 /nimbus-eth1/tools/txparse/txparse.nim(25, 7) Error: type mismatch
#12 7.317 Expression: validate(tx, FkLondon)
#12 7.317   [1] tx: Transaction
#12 7.317   [2] FkLondon: EVMFork
#12 7.317 
#12 7.317 Expected one of (first mismatch at [position]):
#12 7.317 [1] func validate(self: Rlp)
#12 7.317 
#12 7.359 make: *** [Makefile:377: txparse] Error 1
```

Also, it would be nice if you updated the fork to be Prague, on https://github.com/status-im/nimbus-eth1/blob/master/tools/txparse/txparse.nim#L25, while you're at it.

--- a/tests/test_tools_build.nim
+++ b/tests/test_tools_build.nim
@@ -20,4 +20,6 @@ import
   ../tools/t8n/t8n_test,
   ../tools/t8n/t8n_debug,
   ../tools/evmstate/evmstate,
-  ../tools/evmstate/evmstate_test
+  ../tools/evmstate/evmstate_test,
+  ../tools/txparse/txparse,
+  ../tools/txparse/txparse_test

--- a/tools/txparse/config.nims
+++ b/tools/txparse/config.nims
@@ -1,0 +1,12 @@
+# Nimbus
+# Copyright (c) 2022-2025 Status Research & Development GmbH
+# Licensed under either of
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE) or
+#    http://www.apache.org/licenses/LICENSE-2.0)
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT) or
+#    http://opensource.org/licenses/MIT)
+# at your option. This file may not be copied, modified, or distributed except
+# according to those terms.
+
+# chronicles log output will intefere with txparse ouput
+switch("define", "chronicles_enabled=off")

--- a/tools/txparse/txparse.nim
+++ b/tools/txparse/txparse.nim
@@ -12,17 +12,21 @@ import
   eth/[common, rlp],
   stew/byteutils,
   eth/common/transaction_utils,
+  ../common/helpers,
   ../../execution_chain/transaction,
-  ../../execution_chain/common/evmforks
+  ../../execution_chain/core/validate,
+  ../../execution_chain/common/evmforks,
+  ../../execution_chain/common/common
 
-proc parseTx(hexLine: string) =
+proc parseTx(com: CommonRef, hexLine: string) =
   try:
     let
       bytes = hexToSeqByte(hexLine)
       tx = decodeTx(bytes)
       address = tx.recoverSender().expect("valid signature")
 
-    tx.validate(FkLondon)
+    validateTxBasic(com, tx, FkPrague).isOkOr:
+      echo "err: ", error
 
     # everything ok
     echo "0x", address.toHex
@@ -37,7 +41,12 @@ proc parseTx(hexLine: string) =
     echo "err: malformed rlp"
 
 proc main() =
+  let
+    memDB  = newCoreDbRef DefaultDbMemory
+    config = getChainConfig("Prague")
+    com    = CommonRef.new(memDB, nil, config)
+
   for hexLine in stdin.lines:
-    parseTx(hexLine)
+    com.parseTx(hexLine)
 
 main()

--- a/tools/txparse/txparse_test.nim
+++ b/tools/txparse/txparse_test.nim
@@ -1,5 +1,5 @@
 # Nimbus
-# Copyright (c) 2022-2023 Status Research & Development GmbH
+# Copyright (c) 2022-2025 Status Research & Development GmbH
 # Licensed under either of
 #  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE) or
 #    http://www.apache.org/licenses/LICENSE-2.0)
@@ -74,7 +74,7 @@ proc runTest(fileName: string): bool =
 
 proc collectFileNames(inputPath: string, fileNames: var seq[TestFile]) =
   for filename in walkDirRec(inputPath):
-    if not fileName.endsWith(".json"):
+    if not filename.endsWith(".json"):
       continue
 
     fileNames.add TestFile(
@@ -88,6 +88,7 @@ proc main() =
     collectFileNames(testData, filenames)
 
     for input in filenames:
+      let input = input # Workaround compiler lent issue
       test input.dispName:
         let res = runTest(input.fullPath)
         check true == res

